rskj-core/src/test/java/co/rsk/peg/bitcoin/ErpRedeemScriptTestUtils.java

```diff
@@ -0,0 +1,184 @@
+        int pubKey = key.getPubKey().length;
+        return pubKeysLengthByte + pubKey;
+    }
+
+    public static int calculateCSVValueLength(long csvValue) {
+        int opElse = 1;
+        int csvLengthByte = 1;
+        int csvLength = BigInteger.valueOf(csvValue).toByteArray().length;
+        int opCheckSequenceVerify = 1;
+        int opDrop = 1;
+        return opElse + csvLengthByte + csvLength + opCheckSequenceVerify + opDrop;
+    }
+
+    public static int calculateCustomRedeemScriptLength(List<BtcECKey> keys) {
+        int opNotIf = 1;
+        byte[] publicKey = keys.get(0).getPubKey();
+        int opCheckSig = 1;
+        int pubKeyLengthByte = 1;
+        int pubKey = publicKey.length;
+        int pubKeyAndCheckSigLength = pubKeyLengthByte + pubKey + opCheckSig;
+        int customRedeemScriptLength = opNotIf + pubKeyAndCheckSigLength;
+
+        int opSwap = 1;
+        int opAdd = 1;
+        customRedeemScriptLength += (opSwap + pubKeyAndCheckSigLength + opAdd) * (keys.size() - 1);
+
+        int threshold = 1;
+        int opNumEqual = 1;
+        customRedeemScriptLength += (threshold + opNumEqual);
+
+        return customRedeemScriptLength;
+    }
+
+    public static void assertCustomERPRedeemScript(byte[] p2shp2wshErpCustomRedeemScriptProgram, List<BtcECKey> defaultKeys) {
+        /*
+         * Retrieving the federation public keys returns them in lexicographical order. The same happens with the
+         * corresponding signatures. However, the signatures provided are in reverse order than the public keys
+         * in the redeem script. Therefore, we are pushing the keys in reverse order to keep the signatures unmodified.
+         * This is because when evaluating the script, the push operations are added at the bottom of the stack and
+         * compared against the last element in it (i.e., the one that is right above the operation). So the first
+         * pushed operation will be compared against the bottom element of the stack.
+         */
+        List<BtcECKey> reversedDefaultKeys = Lists.reverse(defaultKeys);
+        int defaultThreshold = defaultKeys.size() / 2 + 1;
+
+        // redeemScript - First opcode should be the OP_NOTIF
+        int programCounter = 0;
+        byte actualOpCode = p2shp2wshErpCustomRedeemScriptProgram[programCounter++];
+        assertEquals((byte) ScriptOpCodes.OP_NOTIF, actualOpCode);
+
+        // defaultCustomRedeemScript - First bytes should be the PubKey and OP_CHECKSIG
+        // It checks the first pubKey separately, since it is not preceded by an OP_SWAP
+        BtcECKey expectedPubKey = reversedDefaultKeys.get(0);
+        int pubKeyEnd = programCounter + calculatePubKeyBytesLength(expectedPubKey);
+        byte[] expectedPubKeyBytes = expectedPubKey.getPubKey();
+        byte[] pubKeyProgram = Arrays.copyOfRange(p2shp2wshErpCustomRedeemScriptProgram, programCounter, pubKeyEnd);
+        assertPublicKeyInScript(pubKeyProgram, expectedPubKeyBytes);
```

Comment:
```suggestion
        BtcECKey expectedPubKey = reversedDefaultKeys.get(0);
        int pubKeyEnd = programCounter + calculatePubKeyBytesLength(expectedPubKey);
        byte[] pubKeyProgram = Arrays.copyOfRange(p2shp2wshErpCustomRedeemScriptProgram, programCounter, pubKeyEnd);
        assertPublicKeyInScript(pubKeyProgram, expectedPubKey.getPubKey());
```

super nit, but i find this easier to follow